GitCommands/Git/GitModule.cs

```diff
@@ -1612,15 +1612,11 @@ private string GetFetchArgs(string remote, string remoteBranch, string localBran
                 remoteBranchArguments = "+" + FormatBranchName(remoteBranch);
             }
 
-            string localBranchArguments;
+            string localBranchArguments = "";
             var remoteUrl = GetSetting(string.Format(SettingKeyString.RemoteUrl, remote));
 
             if (PathIsUrl(remote) && !string.IsNullOrEmpty(localBranch) && string.IsNullOrEmpty(remoteUrl))
                 localBranchArguments = ":" + GitCommandHelpers.GetFullBranchName(localBranch);
-            else if (string.IsNullOrEmpty(localBranch) || PathIsUrl(remote) || string.IsNullOrEmpty(remoteUrl))
```

Comment:
Passing null from PullCmd we will go to 1621 line and achieve the desired behavior. Discussion on how the fetch should act should be a subject of another PR in order to not block this PR.

Reply:
Done